pkg/unvanquished_src.dpkdir/bots/attack.bt

```diff
@@ -0,0 +1,110 @@
+selectClass
+{
+	selector
+	{
+		condition team == TEAM_ALIENS
+		{
+			spawnAs PCL_ALIEN_LEVEL0
+		}
+		condition team == TEAM_HUMANS
+		{
+			spawnAs WP_MACHINEGUN
+		}
+	}
+}
+
+selector
+{
+	// reset the suicide timer
+	decorator return( STATUS_FAILURE )
+	{
+		action resetMyTimer
```

Comment:
The suicide timer is `stuckTime`, not `myTimer`. "My timer" is not used anywhere in this BT so this seems pointless.

Reply:
This is the general purpose timer. It is not set or read in C++. But `default.bt` uses it to wait for 20 seconds before suiciding to spawn as builder. If you change a bot from `attack.bt` to `default.bt`, it might suicide immediately if this timer has not been reset in a while.

We should do this in all behavior intented to be available to regular users.

Reply:
Oh so old state is not cleared when switching behaviors? We should fix that.

Reply:
I did not think about this problem when I added that timer. All I cared about was a timer independent of the paths taken in the behavior tree (`decorator timer` is closely tied to these paths and success/fail).

We could reset it on behavior changes. Indeed, we should reset bot targets on behavior change too. But what if someone wants to use that timer in an "inverted" way? Like setting it when some positive condition is met, lets say when `alertedToEnemy`. And later query that time to find out how long a bot has not sensed an enemy. Automatic reset would break this instead.

Reply:
> I did not think about this problem when I added that timer. All I cared about was a timer independent of the paths taken in the behavior tree (`decorator timer` is closely tied to these paths and success/fail).
> 
> We could reset it on behavior changes. Indeed, we should reset bot targets on behavior change too. But what if someone wants to use that timer in an "inverted" way? Like setting it when some positive condition is met, lets say when `alertedToEnemy`. And later query that time to find out how long a bot has not sensed an enemy. Automatic reset would break this instead.

I guess the appropriate way to reset `myTimer` is to set it to the time that the behavior tree was set, rather than 0. Then it would behave the most similarly to as if the game had just started.

Reply:
I opened #2971 to fix this.

Reply:
> I guess the appropriate way to reset `myTimer` is to set it to the time that the behavior tree was set, rather than 0. Then it would behave the most similarly to as if the game had just started.

That seems to be the most sensible choice.